### ISSUE TITLE
Error while fetching R2 file

### ISSUE THREAD

**Issue by lkwr** [skymethod/denoflare#19]
Hello,

a few days ago I created a simple CDN with denoflare and cloudflare r2. But today the r2 request failed with the error: `Error servicing request Error: Expected content-length header`

```
Error servicing request Error: Expected content-length header
    at getExpectedHeader (https://raw.githubusercontent.com/skymethod/denoflare/v0.5.3/cli/api_r2_bucket.ts:180:11)
    at new HeadersBasedR2Object (https://raw.githubusercontent.com/skymethod/denoflare/v0.5.3/cli/api_r2_bucket.ts:236:22)
    at new ResponseBasedR2ObjectBody (https://raw.githubusercontent.com/skymethod/denoflare/v0.5.3/cli/api_r2_bucket.ts:268:9)
    at ApiR2Bucket.get (https://raw.githubusercontent.com/skymethod/denoflare/v0.5.3/cli/api_r2_bucket.ts:61:16)
    at async https://raw.githubusercontent.com/skymethod/denoflare/v0.5.3/common/rpc_r2_bucket.ts:39:67
    at async RpcChannel.receiveMessage (https://raw.githubusercontent.com/skymethod/denoflare/v0.5.3/common/rpc_channel.ts:41:36)
    at async Worker.worker.onmessage (https://raw.githubusercontent.com/skymethod/denoflare/v0.5.3/cli/worker_manager.ts:67:17)
```

This error is only happen when the requested file actually exists on the r2 bucket.

I tried to replace the api key with a new one but the error still happen.

Is this a known issue or am I doing something wrong?

main.ts

```typescript
export default {
  fetch: async (
    req: IncomingRequestCf,
    env: WorkerEnvironment,
    ctx: ModuleWorkerContext
  ): Promise<Response> => {
    const file = await env.bucket.get('assets/logo.svg');
    return new Response('ok');
  },
};

export interface WorkerEnvironment {
  bucket: R2Bucket;
}
```

.denoflare

```
{
  "$schema": "https://raw.githubusercontent.com/skymethod/denoflare/v0.5.3/common/config.schema.json",
  "scripts": {
    "cdn": {
      "path": "./modules/cdn/main.ts",
      "localPort": 8081,
      "bindings": {
        "bucket": {
          "bucketName": "storage"
        }
      }
    }
  },
  "profiles": {
    "default": {
      "default": true,
      "accountId": "xxx",
      "apiToken": "xxx"
    }
  }
}

```

**Comment by johnspurlock-skymethod** [comment]
I can take a look - this happens only when testing locally with `denoflare serve`, right?  That should be the only time the `ApiR2Bucket` proxy is in play.

Also those credentials you use in the config should have R2 permissions granted, just in case that's it.

**Comment by johnspurlock-skymethod** [comment]
Yea I'm seeing it too - it appears something's changed in R2's `HEAD object` S3 api call response.

They used to return `content-length`, but now return `content-range` (and no `content-length`) even for non-range requests.

`content-range` includes the length, so I can parse that out as a quick fix - but I have a question out to their team to see if this was an intentional recent change.

**Comment by johnspurlock-skymethod** [comment]
Unfortunately even `content-range` is not always returned, and other headers are sometimes now missing on R2 S3 api responses, making this proxy difficult (missing content-length, content-range, _and_ etag in some cases)

What's interesting is that these missing headers afflict all items in one bucket, but none of the objects in another.

Do you happen to have additional buckets to test with?  I'd be interested if it happens on some or all of your buckets.  Trying to figure out the underlying pattern.

**Comment by johnspurlock-skymethod** [comment]
Just pushed a commit with some R2 changes to support the latest state of the production R2 S3 api.

Can you try it out?  Substitute the latest commit hash `64451d63a124d15caf764e3292ee34297f14650b` for `v0.5.3` in the denoflare installation instructions.

e.g. if you used deno install, to install to the new commit hash:
```
deno install --unstable --allow-read --allow-net --allow-env --allow-run --name denoflare --force \
https://raw.githubusercontent.com/skymethod/denoflare/64451d63a124d15caf764e3292ee34297f14650b/cli/cli.ts
```

**Comment by lkwr** [comment]
After testing your fix, I got a different error:

```
GET http://localhost:8081/assets/logo.svg
Error servicing request Error: cloneBodyInit(); bodyInit=object [object ReadableStream]
    at cloneBodyInit (blob:null/0dd3a0af-7819-4ea2-8f66-5908a8de66b3:447:11)
    at DenoflareResponse.clone (blob:null/0dd3a0af-7819-4ea2-8f66-5908a8de66b3:409:38)
    at respondAndCache (blob:null/a99fc1ae-af77-40e7-bd5f-8fbe28e5be74:52:47)
    at handleRequest (blob:null/a99fc1ae-af77-40e7-bd5f-8fbe28e5be74:41:20)
    at async Object.fetch (blob:null/a99fc1ae-af77-40e7-bd5f-8fbe28e5be74:83:16)
    at async ModuleWorkerExecution.fetch (blob:null/0dd3a0af-7819-4ea2-8f66-5908a8de66b3:560:16)
    at async WorkerExecution.fetch (blob:null/0dd3a0af-7819-4ea2-8f66-5908a8de66b3:658:16)
    at async blob:null/0dd3a0af-7819-4ea2-8f66-5908a8de66b3:2652:30
    at async RpcChannel.receiveMessage (blob:null/0dd3a0af-7819-4ea2-8f66-5908a8de66b3:39:36)
    at async self.onmessage (blob:null/0dd3a0af-7819-4ea2-8f66-5908a8de66b3:2688:13)
```

**Comment by johnspurlock-skymethod** [comment]
Ok, glad to hear the R2 proxy is working again.

That `cloneBodyInit` error is something that I hadn't implemented yet, namely when running `serve` in isolate mode (the default) and supporting `Response.clone()` when the bodyInit was a `ReadableStream`.  I just implemented that in a091649a98c8c82d2067d4a8076742399d4509bc if you want to go back to using that in your code.

Both of these changes will be in the next release, which I plan on publishing shortly.

**Comment by johnspurlock-skymethod** [comment]
This change is included in release [v0.5.4](https://github.com/skymethod/denoflare/releases/tag/v0.5.4)
